### PR TITLE
[develop2] preliminary support for conf in list-packages

### PR DESCRIPTION
**Pull request by memsharded**

### CODE REVIEW

File: conans/model/info.py [hunk 1]
```diff
@@ -367,9 +370,9 @@ def dumps(self):
         if self.build_requires:
             result.append("[build_requires]")
             result.append(self.build_requires.dumps())
-        conf_dumps = self.conf.dumps()
-        if conf_dumps:
-            # result.append("[conf]")  # Let's do it when apply FIXME above
+        if self.conf:
+            # TODO: Think about the serialization of Conf, not 100% sure if dumps() is the best
+            result.append("[conf]")
```
Comment (franramirez688) [inline conans/model/info.py:375]: I'm not sure what you mean by that `TODO`

File: conans/test/integration/package_id/package_id_and_confs_test.py [hunk 2]
```diff
@@ -37,10 +37,16 @@ class Pkg(ConanFile):
     client.save({"conanfile.py": conanfile, "profile": profile})
     client.run('create . --name=pkg --version=0.1 -s os=Windows -pr profile')
     client.assert_listed_binary({"pkg/0.1": (package_id, "Build")})
+    revision = client.exported_recipe_revision()
+    client.run(f"list packages pkg/0.1#{revision}")
+    if "cxxflags" in package_id_confs:
+        assert "tools.build:cxxflags=['--flag1', '--flag2']" in client.out
+    if "cflags" in package_id_confs:
+        assert "tools.build:cflags=['--flag3', '--flag4']" in client.out
```
Comment (franramirez688) [inline conans/test/integration/package_id/package_id_and_confs_test.py]: I think we could split this test.

**Review by franramirez688 — APPROVED** [review]

File: conans/model/info.py (hunk 1 above)
Comment (memsharded) [reply conans/model/info.py:375]: Conf serializes "Profile-like". Not sure if that is the best serialization to "conaninfo.txt" or we could want to serialize final values instead

File: conans/test/integration/package_id/package_id_and_confs_test.py (hunk 2 above)
Comment (memsharded) [reply conans/test/integration/package_id/package_id_and_confs_test.py]: Done, moved to its own test under ``conan list packages``

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**